stat.Rdev (variable of type uint32) as uint64

**Describe the bug**
[A clear and concise description of what the bug is.]

**To Reproduce**
```go version go1.21.5 windows/amd64
// paste example code reproducing the bug you are reporting
```

**Expected behavior**
[A clear and concise description of what you expected to happen.]

**Environment (please complete the following information):**
 - [ ] Windows: [paste the result of `ver`]
 - [X] Linux: [paste contents of `/etc/os-release` and the result of `uname -a`]
 - [ ] Mac OS: [paste the result of `sw_vers` and `uname -a`
 - [ ] FreeBSD: [paste the result of `freebsd-version -k -r -u` and  `uname -a`]
 - [ ] OpenBSD: [paste the result of `uname -a`]

**Additional context**
[Cross-compiling? Paste the command you are using to cross-compile and the result of the corresponding `go env`]

E:\Go\pkg\mod\github.com\shirou\gopsutil\v3@v3.23.12\disk\disk_linux.go:510:22: cannot use stat.Rdev (variable of type uint32) as uint64 value in argument to unix.Major
E:\Go\pkg\mod\github.com\shirou\gopsutil\v3@v3.23.12\disk\disk_linux.go:511:22: cannot use stat.Rdev (variable of type uint32) as uint64 value in argument to unix.Minor
E:\Go\pkg\mod\github.com\shirou\gopsutil\v3@v3.23.12\disk\disk_linux.go:544:22: cannot use stat.Rdev (variable of type uint32) as uint64 value in argument to unix.Major
E:\Go\pkg\mod\github.com\shirou\gopsutil\v3@v3.23.12\disk\disk_linux.go:545:22: cannot use stat.Rdev (variable of type uint32) as uint64 value in argument to unix.Minor

--- a/Makefile
+++ b/Makefile
@@ -20,6 +20,7 @@ build_test:  ## test only buildable
 	GOOS=linux GOARCH=loong64 go test ./... | $(BUILD_FAIL_PATTERN)
 	GOOS=linux GOARCH=riscv64 go test ./... | $(BUILD_FAIL_PATTERN)
 	GOOS=linux GOARCH=s390x go test ./... | $(BUILD_FAIL_PATTERN)
+	GOOS=linux GOARCH=mips go test ./... | $(BUILD_FAIL_PATTERN)
 	GOOS=freebsd GOARCH=amd64 go test ./... | $(BUILD_FAIL_PATTERN)
 	GOOS=freebsd GOARCH=386 go test ./... | $(BUILD_FAIL_PATTERN)
 	GOOS=freebsd GOARCH=arm go test ./... | $(BUILD_FAIL_PATTERN)

--- a/disk/disk_linux.go
+++ b/disk/disk_linux.go
@@ -507,8 +507,8 @@ func SerialNumberWithContext(ctx context.Context, name string) (string, error) {
 	if err := unix.Stat(name, &stat); err != nil {
 		return "", err
 	}
-	major := unix.Major(stat.Rdev)
-	minor := unix.Minor(stat.Rdev)
+	major := unix.Major(uint64(stat.Rdev))
+	minor := unix.Minor(uint64(stat.Rdev))
 
 	sserial, _ := udevData(ctx, major, minor, "E:ID_SERIAL")
 	if sserial != "" {
@@ -541,8 +541,8 @@ func LabelWithContext(ctx context.Context, name string) (string, error) {
 	if err := unix.Stat(common.HostDevWithContext(ctx, name), &stat); err != nil {
 		return "", err
 	}
-	major := unix.Major(stat.Rdev)
-	minor := unix.Minor(stat.Rdev)
+	major := unix.Major(uint64(stat.Rdev))
+	minor := unix.Minor(uint64(stat.Rdev))
 
 	label, err := udevData(ctx, major, minor, "E:ID_FS_LABEL")
 	if err != nil {

